### PR TITLE
Project refactoring and library upgrades

### PR DESCRIPTION
**Pull request by mrhuber71**
1. Change project character set to UTF-8
2. Refactored POM: All Versions are set as properties
3. Restore Java Service Provider Interface for FOP-Extension in maven build
4. Fix QR-Code functionality with xZing 3.4.1 and later (3.5.1)
5. Increment snapshot version to 2.2.0-SNAPSHOT (latest official version is 2.1.0)
6. Library upgrades:
- Library upgrades
- Apache FOP 2.8
- Batik 1.16
- XML graphics common 2.8
- ZXing 3.5.1
- Xalan 2.7.2
- Xml-APIs 1.4.01
- Avalon Framework 4.3.1
- Apache Commons Cli 1.5

7. Added TAGS to original CSV/SVN commits:
There is no way to make a pull request for tags. So you can add these tags in your repository with following git commands:
`git tag barcode4j_1_0 31223e8095bd0dea7e5545841e816400fe1757f8`
`git tag barcode4j_1_0_rc1 79de4dc97a425cd8bc2730f8cbd4e06d8cb642ef`
`git tag barcode4j_2_0 445b15e626b4c881b2a0740184b15caebb75f986`
`git tag barcode4j_2_0_alpha1 87a42149d1dfc0e88157090794c2d0d543370f38`
`git tag barcode4j_2_0_alpha2 6e24d34c681962ff14664e32f2aa05f498c98d24`
`git tag barcode4j_2_1 274cd4edca0c128ef814e5e7a54e1282d8313e1f`
`git tag before_switch_to_alv2 b3fed673344dd0071e3a259b6dc0e2d68a840fba`

### CODE REVIEW

File: barcode4j/pom.xml [hunk 1]
```diff
@@ -65,55 +60,37 @@
...
-                                <requireJavaVersion>
-                                    <version>[1.8,)</version>
-                                    <message>Java 8 is the minimum supported</message>
-                                </requireJavaVersion>
+                                <requireFilesExist>
+                                    <files>
+                                        <file>${project.parent.basedir}/LICENSE</file>
+                                        <file>${project.parent.basedir}/NOTICE</file>
+                                    </files>
+                                </requireFilesExist>
```
Comment (SingingBush) [inline barcode4j/pom.xml:80]: I'm fine with adding the new rule but why where the require version rules removed?

Comment (mrhuber71) [reply barcode4j/pom.xml:80]: The Rules are not removed. You will find these enforcer rules within pluginManagement section: "requireMavenVersion", "requireMavenVersion" and "requirePluginVersions" in the parent pom.xml.

File: pom.xml [hunk 2]
```diff
@@ -85,177 +86,299 @@
...
+                    <version>${plugin.version.maven-source-plugin}</version>
+                </plugin>
+                <!-- maven reporting plugins -->
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-javadoc-plugin</artifactId>
+                    <version>${plugin.version.maven-javadoc-plugin}</version>
+                </plugin>
+                <!-- maven tools plugins -->
+                <plugin>
```
Comment (mrhuber71) [inline pom.xml:304]: Here you can see the enforcer rules in reactor pom. these rules are common to all modules.

**Review by SingingBush — APPROVED** [review]

**Comment by mrhuber71** [comment]
Hi,
thanks for accepting pull request :-)
Other question: Is it possible for you to add the tags from the original source SVN-Repository?
See initial comment on pull request.

Mathias

**Comment by SingingBush** [comment]
yeah, I ran them on my pc this morning but forgot to push them. Will sort it once I'm home. Do you intend to make any further changes? I no longer use JDK 8 as Java 17 is my target for pretty much all projects. I'm considering bumping the minimum JDK to 11 so that I can add JPMS support. Would that be a problem for you?

**Comment by mrhuber71** [comment]
@tags -> very nice 👍 

Java 11/17 is OK because we already use Java 17 for the most part in our company. I also think that the JPMS support will not be a problem for us.

We don't have any expansion plans at the moment, but I can imagine that we'll certainly do bug fixes.

It would be nice if you manage (we? manage) to bring the releases to Maven Central. But there is certainly still one or the other to do (e.g. Ant versus Maven build is not the same).

Ant versus Maven build:
What still needs to be done is for the Maven build to create the same JARs as the Ant build. For example, I noticed that the FOP extension does not contain the classes from the "barcode-xgc" module. These are necessary to bring the barcodes as images into the PDFs. In our projects, the JAR for the "barcode-xgc" module is also used (and so it is not a direct problem). For the FOP extension I think the easiest way would be to combine the "barcode-xgc" module with the "barcode4j-fop-ext" module.

**Comment by SingingBush** [comment]
I just checked maven central. Seems 2.1 just had two artifacts:

- barcode4j
- barcode4j-fop-ext

so bundling xgc into fop-ext seems reasonable. To be honest I've not used this repo for a while. The barcode4j dependency was used heavily by my previous employer and so I was looking to revive the project just for my own sanity. I don't need to generate labels for my current line of work but am still interested in keeping barcode4j alive and can publish to maven central as well, although that will likely have to be using my own groupId: [com.singinbush](https://search.maven.org/search?q=com.singingbush). I think it would be wise to release a 2.2.0 version to maven central that has the same _org.krysalis.barcode4j_ package name so that there are no breaking changes for anyone wanting to use it. 

Before doing that it would be good to get some thorough testing done, is that something you can help with?

**Comment by mrhuber71** [comment]
May I help you to do the merge of these two Moduls in a new pull request?

@ groupId:
As you say, I think for this project we should use "org.krysalis.barcode4j", yes.
Should I try to conatct the initiator Jeremias Märki of the project if it is OK to use this groupId ( Home page https://www.jeremias-maerki.ch/wordpress/de/software/ )?

@ testing
Maybe I can help you to move tests to JUnit 5.

**Pull request merged**